src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressByMessageFilterTest.java

```diff
@@ -0,0 +1,162 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.filters;
+
+import static com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck.MSG_FILE_CONTAINS_TAB;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+public class SuppressByMessageFilterTest extends AbstractModuleTestSupport {
+
+    private static final String[] ALL_MESSAGES = {
+        "16:1: " + getCheckMessage(FileTabCharacterCheck.class, MSG_FILE_CONTAINS_TAB),
+    };
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/puppycrawl/tools/checkstyle/filters/suppressbymessagefilter";
+    }
+
+    @Test
+    public void testDefault() throws Exception {
+        final String[] suppressed = CommonUtil.EMPTY_STRING_ARRAY;
+        verifySuppressedWithParser(getPath("InputSuppressByMessageFilter.java"), suppressed);
+    }
+
+//    @Test
+//    public void testMatchNothing() throws Exception {
+//        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+//        config.addProperty("checkFormat", "");
+//        config.addProperty("messageFormat", "");
```

Comment:
please remove such methods, you should use test methods that use verifySuppressedWithParser

Look at   SuppressWithNearbyCommentFilterTest is testing Filter that is laso under Checker.

Reply:
This is the past work done before BDD, please ignore for review.